### PR TITLE
Add unions

### PR DESCRIPTION
**Pull request by rafalp**
This PR adds support for `union` types in GraphQL.

This is achieved by introduction of `ariadne.Union` bindable that takes name of union, and resolver function that takes any object and is expected to return `None` or `str` with GraphQL type name.

`Union` also implements new API as discussed in #98:

```
union = Union("Name", type_resolver)

# ...or...

@union.type_resolver
def resolver(*_):
    ...
```

### CODE REVIEW

File: tests/test_unions.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                username
+            }
+            ... on Thread {
+                title
+            }
+        }
+    }
+"""
+
+user = Mock(username="User")
```
Comment (salwator) [inline tests/test_unions.py]: Naming module-scoped symbols should be different then local variables, especially when we have same naming convention for pytest fixtures.
I would make it _USER_ or _User_ at least. Same applies for thread.

File: tests/test_unions.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    result = graphql_sync(schema, "{ item { __typename } }")
+    assert result.data == {"item": {"__typename": "User"}}
+
+
+def resolve_result_type(obj, *_):
+    if obj == user:
+        return "User"
+    if obj == thread:
+        return "Thread"
+    return None
```
Comment (salwator) [inline tests/test_unions.py]: No need for that explicit return.

File: tests/test_unions.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    union = Union("FeedItem", type_resolver=resolve_result_type)
+
+    schema = make_executable_schema(type_defs, [query, union])
+    result = graphql_sync(schema, test_query)
+    assert result.data == {"item": {"__typename": "Thread", "title": thread.title}}
+
+
+def test_result_is_none_if_union_didnt_resolve_the_type():
+    query = ResolverMap("Query")
+    query.field(  # pylint: disable=unexpected-keyword-arg
```
Comment (salwator) [inline tests/test_unions.py]: We might want to just disable warning for tests, for example by local pylintrc file.

Comment (salwator) [reply tests/test_unions.py]: Or we can just `# pragma pylint: disable=unexpected-keyword-arg` at the top.

File: tests/test_unions.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    if obj == user:
+        return "User"
+    if obj == thread:
+        return "Thread"
+    return None
+
+
+def test_result_is_username_if_union_resolves_type_to_user():
+    query = ResolverMap("Query")
+    query.field(  # pylint: disable=unexpected-keyword-arg
```
Comment (salwator) [inline tests/test_unions.py]: Resolver map with configured field is duplicated in test cases, it is perfect case for pytest fixture.

File: tests/test_unions.py (hunk 3 above)
Comment (rafalp) [reply tests/test_unions.py]: I agree that code littered with comments disabling linter looks bad, but I think it's more explicit and safer to work with than one that blanket-disables everything.

I'm also using those as a reminder that our current `ResolverMap` API confuses the hell out of `PyLint` and we really need to change it, but that's tracked in #98

File: tests/test_unions.py (hunk 2 above)
Comment (rafalp) [reply tests/test_unions.py]: This code is here to satisfy pylint `inconsistent-return-statements`.

**Pull request merged**